packages/core/src/utils.ts

```diff
@@ -259,6 +263,137 @@ export async function* maskStreamTags(
   }
 }
 
+/**
+ * Resolve serialized zod output - This function takes the string output ot the `jsonSchemaToZod` function
+ * and instantiates the zod object correctly.
+ *
+ * @param obj - serialized zod object
+ * @returns resolved zod object
+ */
+export function resolveSerializedZodOutput(obj: any) {
+  return Function('z', `"use strict";return (${obj});`)(z);
```

Comment:
Would be good to add a comment here about how this works

Reply:
Added a comment